File: frontend/src/app/core/models/task.dto.ts
import { SimpleUserDto } from './user.dto';

export interface TaskDto {
  id: number;
  boardId: number;
  columnId: number;
  title: string;
  description: string;
  priority: string; // LOW, MEDIUM, HIGH, URGENT
  status: string; // ACTIVE, PENDING_APPROVAL
  position: number;
  assignee?: SimpleUserDto;
  createdBy: SimpleUserDto;
  dueDate?: string;
  tags?: string[];
  rejectionReason?: string;
  version: number;
  createdAt: string;
  updatedAt: string;
}

export interface TaskMoveRequest {
  targetColumnId: number;
  newPosition: number;
  adminBypass: boolean;
  version: number;
}

export interface TaskMetadataRequest {
  title: string;
  description?: string;
  priority?: string;
  dueDate?: string;
  tags?: string[]; // Symmetrical array of strings aligned with backend update schema
  version: number;
}

export interface TaskAssigneeRequest {
  assigneeId?: number; // can be null to unassign
  version: number;
}

export interface TaskApproveRequest {
  version: number;
}

export interface TaskRejectRequest {
  fallbackColumnId: number;
  rejectionReason: string;
  version: number;
}

export interface TaskCreateRequest {
  boardId: number;
  columnId: number;
  title: string;
  description?: string;
  priority?: string;
  position: number;
  assigneeId?: number;
  dueDate?: string;
  tags?: string[];
}

File: backend/src/main/java/com/valeo/kanban/dto/response/TaskDto.java
package com.valeo.kanban.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.Instant;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TaskDto {
    private Long id;
    private Long boardId;
    private Long columnId;
    private String title;
    private String description;
    private String priority;
    private String status;
    private double position;
    private SimpleUserDto assignee;
    private SimpleUserDto createdBy;
    private Instant dueDate;
    private List<String> tags;
    private String rejectionReason;
    private Long version;
    private Instant createdAt;
    private Instant updatedAt;

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class SimpleUserDto {
        private Long id;
        private String email;
        private String firstName;
        private String lastName;
    }
}
